runners/samza/src/main/java/org/apache/beam/runners/samza/translation/TranslationContext.java

```diff
@@ -145,9 +150,30 @@ protected <KeyT, OutT> void registerInputMessageStreams(
   }
 
   public <OutT> void registerMessageStream(PValue pvalue, MessageStream<OpMessage<OutT>> stream) {
+    registerMessageStream(pvalue, stream, true);
+  }
+
+  public <OutT> void registerMessageStreamWithoutMetricOp(
+      PValue pvalue, MessageStream<OpMessage<OutT>> stream) {
+    registerMessageStream(pvalue, stream, false);
```

Comment:
Please remove this boolean flag. We should attach the MetricOp if the TransformMetric is enabled. But inside the op, we can decide what to do with it.

Reply:
done